virttest/utils_test/libvirt.py

```diff
@@ -1159,6 +1159,54 @@ def thread_func_migration(self, vm, desturi, options=None,
                 self.RET_MIGRATION = False
                 self.RET_LOCK.release()
 
+    def migrate_pre_setup(self, desturi, params=None, cleanup=False):
+        """
+        # Setup before migration,
+        # 1. To enable migration ports using iptables
+        # 2. Turn off SMT for power8 machine in remote machine to migrate
```

Comment:
@will-Do , sure
1. But I added a new method migrate_pre_setup()  here is all the files virsh_migrate.py, migrate_vm.py, migrate_multi_vms.py, virsh_migrate_stress.py, virsh_migrate_virtio_scsi.py, virsh_migrate_compcache.py, virsh_migrate_set_get_speed.py requires this setup and this can be called before and after migration for setting up and cleaning up and helps us to avoid redundancy.

2. As this SMT have to be switched off only for ppc where as x86 doesn't have this limitation, For local machine, it takes cares for PPC in test_setup before our test even starts, but for remote destination PPC machine in migration test can be taken care here.

Please correct me if I am wrong.